tests/telex.rs

```diff
@@ -0,0 +1,25 @@
+use std::fmt::Write;
+
+fn snapshot_transform(input: &str) -> String {
```

Comment:
Just one small request before I approve this PR. Could you also make this generic as well? since the logic is the same for both `telex` & `vni`, only the `transform_buffer` function is different.